src/gluonts/model/n_beats/_ensemble.py

```diff
@@ -0,0 +1,366 @@
+    ) -> "NBEATSEnsemblePredictor":
+        predictors = []
+        # load all the predictors individually and also make sure not to load anything else by mistake
+        predictor_locations = [
+            sub_dir
+            for sub_dir in next(os.walk(str(path)))[1]
+            if sub_dir.startswith("predictor_")
+        ]
+
+        # deserialize predictors
+        for sub_dir in predictor_locations:
+            predictors.append(
+                RepresentableBlockPredictor.deserialize(path / sub_dir, ctx)
+            )
+
+        # deserialize constructor parameters
+        with (path / "parameters.json").open("r") as fp:
+            parameters = load_json(fp.read())
+
+        return NBEATSEnsemblePredictor(
+            prediction_length=parameters["prediction_length"],
+            freq=parameters["freq"],
+            predictors=predictors,
+            aggregation_method=parameters["aggregation_method"],
+        )
+
+    # Additionally implemented since we are dealing with RepresentableBlockPredictor
+    def hybridize(self, batch: DataBatch) -> None:
+        for predictor in self.predictors:
+            predictor.hybridize(batch)
+
+    def predict(
+        self, dataset: Dataset, num_samples: Optional[int] = 1, **kwargs
+    ) -> Iterator[Forecast]:
+        if num_samples != 1:
+            logging.warning(
+                "NBEATSEnsemblePredictor does not support sampling. "
+                "Therefore 'num_samples' will be ignored and set to 1."
+            )
+        iterators = []
+
+        # create the iterators from the predictors
+        for predictor in self.predictors:
+            iterators.append(predictor.predict(dataset, num_samples=1))
+
+        # we always have to predict for one series in the dataset with
+        # all models and return it as a 'SampleForecast' so that it is
+        # clear that all these prediction concern the same series
+        for item in dataset:
+            output = []
+            start_date = None
+
+            for iterator in iterators:
+                prediction = next(iterator)
+                output.append(prediction.samples)
+
+                # some weird transformation is going on with the start date,
+                # so I just take the already transformed one, otherwise
+                # evaluator will give Nans for the seasonal_error
+                if start_date is None:
+                    start_date = prediction.start_date
+            output = np.stack(output, axis=0)
+
+            # aggregating output of different models
+            # default according to paper is median,
+            # but we can also make use of not aggregating
+            if self.aggregation_method == "median":
+                output = np.median(output, axis=0)
+            elif self.aggregation_method == "mean":
+                output = np.mean(output, axis=0)
+            else:  # do not aggregate
+                pass
```

Comment:
I looked at the unaggregated forecast results, and it is not trivial to aggregate them later, because the actual forecasts are in a list of dicts. Its just way easier and more convenient to specify the aggregation method as an argument that can be set right before the evaluation. Additionally setting the default the median would be according to the paper specification.

Reply:
Bit mean seems to be doing good too.